packages/framework/attributor/src/test/attribution/sharedString.attribution.spec.ts

```diff
@@ -362,11 +369,11 @@ function spyOnOperations(baseGenerator: Generator<Operation, FuzzTestState>): {
 	return { operations, generator };
 }
 
-function readJson(filepath: string): Jsonable {
-	return JSON.parse(readFileSync(filepath, { encoding: "utf-8" }));
+function readJson<T extends Jsonable<unknown>>(filepath: string): Jsonable<T> {
+	return JSON.parse(readFileSync(filepath, { encoding: "utf-8" })) as Jsonable<T>;
 }
 
-function writeJson(filepath: string, content: Jsonable) {
+function writeJson(filepath: string, content: Jsonable<unknown>) {
```

Comment:
In the past, you would want to use the same pattern as readJson:

function writeJson<T>(..., content: Jsonable<T>)

This would infer (if possible) the concrete type of T and recursively check to make sure it doesn't contain forbidden things.

Is Jsonable<unknown> just as good now?

Reply:
We can write `writeJson<T extends JsonableType>(..., content: Jsonable<T>)` (which I think is the only way that any checking would happen), but I don't think it provides any advantage now over simple `Jsonable<unkown>` aka `JsonableType` content type.

Reply:
I dug into this some more. Using `writeJson<T>(..., content: Jsonable<T>)` but not `writeJson<T extends JsonableType>(..., content: Jsonable<T>)` will get past the interface vs. type alias issue relating to `Index signature for type 'string' is missing in type 'Foo'.ts(2345)`.
However, doing the same to `readJson` (removing the generic parameter constraint) lets things through that are not correct like having a function on the type. This also really reasserts the asymmetry of serializable to and from string.

I'm placing a deserialized JSON filter in this file as it has immediate need. That type should be moved to datastore-definitions when we have a plan for distinguishing to-serialize and from-serialization types.

cc: @CraigMacomber